tests/compile-fail/copies.rs

```diff
@@ -249,6 +249,21 @@ fn if_same_then_else() -> Result<&'static str, ()> {
         -0.0
     };
 
+    // Different NaNs
+    let _ = if true {
+        1.0 / 0.0
+    } else {
+        (-5f32).sqrt()
```

Comment:
Oh, I see. I was getting the values from the link @oli-obk posted and didn't question that they were all NaN. I will update it.